backends/bmv2/run-bmv2-ptf-test.py

```diff
@@ -15,137 +17,157 @@
+PARSER.add_argument("p4_file", help="the p4 file to process")
 PARSER.add_argument(
     "-tf",
     "--testfile",
     dest="testfile",
-    help="Provide the path for the ptf py file for this test. ",
+    help="The path for the ptf py file for this test.",
+)
+PARSER.add_argument(
+    "-td",
+    "--testdir",
+    dest="testdir",
+    help="The location of the test directory. ",
 )
 
 
 def is_port_in_use(port: int) -> bool:
     with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
-        return s.connect_ex(('localhost', port)) == 0
+        return s.connect_ex(("localhost", port)) == 0
 
 
 class Options:
-
     def __init__(self):
-        self.p4Filename = ""  # File that is being compiled
-        self.testfile = ""  # path to ptf test file that is used
+        self.p4_file: Path = None  # File that is being compiled
+        self.testfile: Path = None  # path to ptf test file that is used
         # Actual location of the test framework
-        self.testdir = os.path.dirname(os.path.realpath(__file__))
-        self.jsonName = "test.json"
-        self.infoName = "test.p4info.txt"
-        self.rootDir = "."
+        self.testdir: Path = None
+        self.rootDir: Path = "."
 
 
-def _create_bridge():
-    print(
-        "---------------------- Start creating of bridge ----------------------"
+def create_bridge():
+    testutils.report_output(
+        sys.stdout,
+        "---------------------- Start creating of bridge ----------------------",
     )
     random.seed(datetime.now().timestamp())
     outputs = {}
     outputs["stdout"] = sys.stdout
     outputs["stderr"] = sys.stderr
     bridge = Bridge(random.random(), outputs, True)
     result = bridge.create_virtual_env(8)
-    if result != p4c_utils.SUCCESS:
+    if result != testutils.SUCCESS:
         bridge.ns_del()
-        print(
-            "---------------------- End creating of bridge with errors ----------------------"
+        testutils.report_err(
+            sys.stderr,
+            "---------------------- End creating of bridge with errors ----------------------",
         )
-        exit()
-    print("---------------------- Bridge created ----------------------")
+        sys.exit(1)
+    testutils.report_output(
+        sys.stdout, "---------------------- Bridge created ----------------------"
+    )
     return bridge
 
 
-def _open_proc(bridge):
+def open_proc(bridge):
     proc = bridge.ns_proc_open()
     if not proc:
         bridge.ns_del()
-        print(
-            "---------------------- Unable to open bash process in the namespace ----------------------"
+        testutils.report_err(
+            sys.stderr,
+            "---------------------- Unable to open bash process in the namespace ----------------------",
         )
-        exit()
+        sys.exit(1)
     return proc
 
 
-def _kill_process(proc):
+def kill_process(proc):
     # kill process
     os.kill(proc.pid, 15)
```

Comment:
There should probably be at list a short wait between the two kills, otherwise it is likely to be equivalent to just killing with 9 directly.

Reply:
iirc the purpose of that function was a very aggressive kill, where `proc.kill()` and `proc.terminate() `do not work.

Reply:
I see. Well according to the documentation `proc.kill()` should be really equivalent to sending `SIGKILL = 9` (and `.terminate()` to `SIGTERM = 15`.